apps/session-manager/src/controllers/__tests__/zendesk.spec.ts

```diff
@@ -0,0 +1,244 @@
+import mockReq from "../../__mocks__/request.mocks";
+import { ZendeskController } from "../";
+import {
+  mockedInitializedProfile,
+  mockedUser,
+} from "../../__mocks__/user.mocks";
+import mockRes from "../../__mocks__/response.mocks";
+import { FnAppAPIClient } from "../../repositories/fn-app-api";
+import { mockRedisClientSelector } from "../../__mocks__/redis.mocks";
+import { ProfileService, TokenService } from "../../services";
+
+const mockGetProfile = vi.spyOn(ProfileService, "getProfile");
+
+const mockGetZendeskSupportToken = vi.spyOn(
+  TokenService,
+  "getJwtZendeskSupportToken",
+);
+
+const aZendeskSupportToken =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiTWFyaW8gUm9zc2kiLCJlbWFpbCI6Im1hcmlvLnJvc3NpQHRlc3QuaXQiLCJleHRlcm5hbF9pZCI6IkFBQUFBQUFBQUFBQUFBQSIsImlhdCI6MTYzNTk0NDg2MC43ODksImp0aSI6IjAxRktKWUszM04wRkpKNEVZRTM5NURIWlhDIiwiZXhwIjoxNjM1OTQ0OTIwLCJpc3MiOiJJU1NVRVIifQ.dqQlnKGME5FvpI2GbkVpk7vpgE_ft0IZE3jp2YRWHtA";
+
+describe("ZendeskController#getZendeskSupportToken", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+  const res = mockRes() as unknown as express.Response;
+  const req = mockReq() as unknown as express.Request;
+
+  const mockedDependencies = {
+    // Repositories are not used, since we mocked the service layer
+    fnAppAPIClient: {} as ReturnType<typeof FnAppAPIClient>,
+    redisClientSelector: mockRedisClientSelector,
+    user: mockedUser,
+    req,
+    jwtZendeskSupportTokenSecret: "aSecret" as NonEmptyString,
+    jwtZendeskSupportTokenExpiration: 1200 as Second,
+    jwtZendeskSupportTokenIssuer: "issuer" as NonEmptyString,
+  };
+
+  it("should return a valid Zendesk support token when user has a validated email", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(ResponseSuccessJson(mockedInitializedProfile)),
+    );
+
+    mockGetZendeskSupportToken.mockReturnValue(TE.of(aZendeskSupportToken));
+
+    const response = await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.map((response) => response.apply(res)),
+    )();
+
+    expect(E.isRight(response)).toBeTruthy();
+    expect(res.json).toHaveBeenCalledWith({
+      jwt: aZendeskSupportToken,
+    });
+  });
+
+  it("should return an IResponseErrorInternal if user does not have any email address", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(
+        ResponseSuccessJson({
+          ...mockedInitializedProfile,
+          email: undefined,
+          is_email_validated: false,
+          spid_email: undefined,
+        }),
+      ),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Profile has not a validated email address",
+          ),
+        ),
+      ),
+    )();
+  });
```

Comment:
This test does not fail if `ZendeskController.getZendeskSupportToken` returns a `TE.Right` value.
The issue arises because the test only checks the `TE.mapLeft` flow, missing the scenario where a `TE.Right` value is returned instead.
The same issue is present in other tests.

Reply:
thank you @gquadrati. the issue was solved in https://github.com/pagopa/io-auth-n-identity-domain/pull/34/commits/15b5283dc8882ae73e35c2dc05e3f91746718016